### PR TITLE
subscriber: don't gate `with_ansi()` on the "ansi" feature

### LINKED ISSUE #2996: `SubscriberBuilder::with_ansi` is incorrectly gated by the `ansi` feature in tracing-subscriber 0.3.18
### Version

```
tracing v0.1.40
tracing-subscriber v0.3.18
```

### Platform
```
Linux 6.2.0-39-generic #40~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Nov 16 10:53:04 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

```

### Crates

`tracing-subscriber`

### Description

The `tracing-subscriber` 0.3.18 docs claim that `SubscriberBuilder::with_ansi()` can be used even when the `ansi` feature is disabled. And indeed this is true on master. But in the released 0.3.18 code this is not true: commit 049ad730c did not remove the `#[cfg(feature = "ansi")]` from the method.

The 0.3.18 version of the code is here: https://github.com/tokio-rs/tracing/blob/8b7a1dde69797b33ecfa20da71e72eb5e61f0b25/tracing-subscriber/src/fmt/mod.rs#L623-L637

### PR DESCRIPTION
**Pull request by eric-seppanen**
The commit 1cb523b87d3d removed this cfg gate on master. However, when the change was backported in 049ad730c1ba the docs were updated but the `cfg` change was omitted. Some more detail and links are in issue #2996.

This made the docs misleading, since they say "This method itself is still available without the feature flag."

Fixes #2996.

### CODE REVIEW

**Comment by eric-seppanen** [comment]
I'm not sure if this escaped notice, but it looks like this bug was introduced by an error during backporting. I'm not trying to make policy decisions for the stable branch, just trying to finish the job that was started by @hawkw in #2570.

**Review by hawkw — APPROVED** [review]
Thank you for catching this!

**Comment by eric-seppanen** [comment]
Github actions aren't starting. I'm guessing that's because this PR is more than 30 days old. I'll try a no-op push to the branch to see if it gets things moving.

**Comment by eric-seppanen** [comment]
Ah, now it's stopped by "This workflow requires approval from a maintainer." @hawkw can you help?

Also, auto-merge was disabled because I pushed to the branch. (No changes were made.)

**Comment by eric-seppanen** [comment]
Ping? @hawkw you approved this PR but CI won't run until a maintainer intervenes.

**Comment by davidbarsky** [comment]
My bad! Running the workflow now.

**Comment by eric-seppanen** [comment]
Clippy lint warnings appear to be unrelated; CI runs latest stable and at least some of the lints are things I recognize as being new in 1.80.

**Comment by eric-seppanen** [comment]
CI is once again blocked on "This workflow requires approval from a maintainer". Could someone help move it along?

As far as I can tell, this is ready to merge. It's a very simple fix and received approval months ago.

**Comment by eric-seppanen** [comment]
Could someone please approve the CI workflow for this PR? As seen earlier, if it gets more than 30 days old then CI won't be able to run.

**Comment by eric-seppanen** [comment]
Rebased again on the latest v0.1.x branch. There were no conflicts. This PR still needs manual approval for CI to run.

**Pull request merged**
